Add new ref-resolve docu page and deprecate old ref-resolve (#184)
eclipsesource/jsonforms2-website: content/docs/ref-resolving.mdx

:::caution ATTENTION

With version 3.0 of JSON Forms, we no longer include `json-schema-ref-parser` within the core page.
The old documentation for JSON Forms < 3.0 can be found <a href="/docs/deprecated/ref-resolving-legacy">here</a>.
For a migration guide, have a look <a href="https://github.com/eclipsesource/jsonforms/blob/master/MIGRATION.md">here</a>.

:::

Resolving of external dependencies is not a core feature of JSON Forms which is why JSON Forms itself can only handle simple `$ref` resolving, which should be sufficient in most cases.

> For a more complex scenario or when resolving references pointing to external schemas, you can use `json-schema-ref-parser` of a similar library to resolve references before passing them to JSON Forms.

Reviewer: ```suggestion
JSON Forms is able to resolve basic JSON Schema `$ref` pointers which is sufficient for most use cases.
In case of complex reference setups or references pointing to external resources the schema needs to be resolved before handing it over to JSON Forms.
We recommend using a library like `json-refs` or `json-schema-ref-parser` for these use cases.
```

Reviewer: Please also link to the documentation for each library

> Below you can find an example, on how to use `json-schema-ref-parser`:

Reviewer: ```suggestion
Below you can find an example on how to use `json-refs` and `json-schema-ref-parser`:
```

```ts
import React, { useState } from 'react';
import { JsonForms } from '@jsonforms/react';
import { materialCells, materialRenderers } from '@jsonforms/material-renderers';
import $RefParser from '@apidevtools/json-schema-ref-parser';
import JsonRefs from 'json-refs';

import mySchemaWithReferences from 'myschema.json';

const refParserOptions = {
  dereference: {
    circular: false
  }
}

function App() {
  const [data, setData] = useState(initialData);
  const [resolvedSchema, setSchema] = useState();

  useEffect(() => {

>     $RefParser.dereference(mySchemaWithReferences).then(res => setSchema(res.$schema));

Reviewer: You're missing the refParserOptions here. Please check whether the format defined above is the correct one for this call

Author: Seems to be correct: https://apitools.dev/json-schema-ref-parser/docs/options.html#example

:::caution ATTENTION

With version 3.0 of JSON Forms, we no longer include `json-schema-ref-parser` within the core page.
The old documentation for JSON Forms < 3.0 can be found <a href="/docs/deprecated/ref-resolving-legacy">here</a>.
For a migration guide, have a look [here](https://github.com/eclipsesource/jsonforms/blob/master/MIGRATION.md).

:::

JSON Forms is able to resolve basic JSON Schema `$ref` pointers which is sufficient for most use cases.
In case of complex reference setups or references pointing to external resources the schema needs to be resolved before handing it over to JSON Forms.

> We recommend using a library like `json-refs` ([Documentation](https://github.com/whitlockjc/json-refs/tree/master/docs)) or `json-schema-ref-parser` ([Documentation](https://apitools.dev/json-schema-ref-parser/)) for these use cases.

Reviewer: ```suggestion
We recommend using a library like [`json-refs`](https://github.com/whitlockjc/json-refs/tree/master/docs) or [`json-schema-ref-parser`](https://apitools.dev/json-schema-ref-parser/) for these use cases.
```

Below you can find an example on how to use `json-refs` and `json-schema-ref-parser`:

```ts
import React, { useState } from 'react';
import { JsonForms } from '@jsonforms/react';
import { materialCells, materialRenderers } from '@jsonforms/material-renderers';
import $RefParser from '@apidevtools/json-schema-ref-parser';
import JsonRefs from 'json-refs';

import mySchemaWithReferences from 'myschema.json';

> const refParserOptions = {

Reviewer: ```suggestion
const yourRemoteSchemaResolver = {
  order: 1,
  canRead: function(file) {
    return file.url.indexOf('yourRemoteSchemaIdentifier') !== -1;
  },
  read: function() {
    return JSON.stringify(yourSchemaObject);
  },
};
const refParserOptions = {
```

>   dereference: {
>     circular: false
>   }
> }

Reviewer: ```suggestion
  dereference: {
    circular: false
  },
  resolve: {
    foo: yourRemoteSchemaResolver
}
```

:::caution ATTENTION

> With version 3.0 of JSON Forms, we no longer include `json-schema-ref-parser` within the core page.

Reviewer: ```suggestion
With version 3.0 of JSON Forms, we no longer include `json-schema-ref-parser` within the core package.
```
